app/controllers/application_controller.rb

```diff
@@ -1,14 +1,43 @@
 class ApplicationController < ActionController::API
-  def current_user
-    @current_user || authenticate_token
+  include ActionController::HttpAuthentication::Token::ControllerMethods
+
+  attr_reader :current_user
+  before_action :authenticate
+
+private
+  def authenticate
+    authenticate_token || unauthorized_token
   end
 
   def authenticate_token
-    payload, _header = TokenManager.authenticate(request)
-    if payload
-      @current_user = User.find_by(id: payload["user"])
-      return @current_user if @current_user && @current_user.active
+    authenticate_with_http_token do |auth_token, options|
+      payload, _header = TokenManager.decode(auth_token)
+      user_id = payload ? payload['user'] : nil
+      @current_user = User.find_by(id: user_id)
```

Comment:
Nice rewrite. Cool.
We can also go further and modify `TokenManager.decode` method to return just the `payload` since we don't really need the header. 
Then we can modify this method to be something like:
```ruby
payload = TokenManager.decode(auth_token)
@current_user = User.find_by(id: payload["user']) if payload
```